spec.bs

```diff
@@ -309,6 +308,101 @@ The following are the [=event handlers=] (and their corresponding [=event handle
+    * {{AppHistoryNavigationOptions/navigateInfo}} can be set to any value; it will populate the {{AppHistoryNavigateEvent/info}} property of the corresponding {{AppHistory/navigate}} event.
+    * {{AppHistoryNavigateOptions/state}} can be set to any value; it will populate the state retrieved by {{AppHistoryEntry/getState()|appHistory.current.getState()}} once the navigation completes, for same-document navigations. (It will be ignored for navigations that end up cross-document.)
+
+    <p>By default this will perform a full navigation (i.e., a cross-document navigation, unless the given URL differs only in a fragment from the current one). The {{AppHistory/navigate}} event's {{AppHistoryNavigateEvent/respondWith()}} method can be used to convert it into a same-document navigation.
+
+    <p>The returned promise will behave as follows:
+
+    * For same-document navigations created by using the {{AppHistory/navigate}} event's {{AppHistoryNavigateEvent/respondWith()}} method, it will fulfill or reject according to the promise passed to {{AppHistoryNavigateEvent/respondWith()}}.
+    * For other same-document navigations (e.g., non-intercepted <a spec="HTML" lt="navigate to a fragment">fragment navigations</a>), it will fulfill immediately.
+    * For cross-document navigations, it will never settle.
+  </dd>
+
+  <dt><code>await {{Window/appHistory}} . {{AppHistory/navigate(options)|navigate}}(<var ignore>options</var>)</code>
+  <dd>
+    <p>Navigates to the same URL as the current page. <var ignore>options</var> can contain {{AppHistoryNavigationOptions/navigateInfo}} and {{AppHistoryNavigateOptions/state}} which behave as described above.
+
+    <p>The default behavior of performing a full navigation to the current page can be overriden by using the {{AppHistory/navigate}} event's {{AppHistoryNavigateEvent/respondWith()}} method. Doing so will mean this call only updates state or passes along the appropriate {{AppHistoryNavigationOptions/navigateInfo}}.
+
+    <p>The returned promise behaves as described above.
+  </dd>
+</dl>
+
+<div algorithm>
+  The <dfn method for="AppHistory">navigate(|url|, |options|)</dfn> method steps are:
+
+  1. <a spec="HTML" lt="parse a URL">Parse</a> |url| relative to [=this=]'s [=relevant settings object=]. If that returns failure, then return [=a promise rejected with=] a "{{SyntaxError}}" {{DOMException}}. Otherwise, let |urlRecord| be the <a spec="HTML">resulting URL record</a>.
+
+  1. Return the result of [=performing an app history navigation=] given [=this=], |urlRecord|, and |options|.
+</div>
+
+<div algorithm>
+  The <dfn method for="AppHistory">navigate(|options|)</dfn> method steps are:
+
+  1. If neither |options|["{{AppHistoryNavigationOptions/navigateInfo}}"] nor |options|["{{AppHistoryNavigateOptions/state}}"] [=map/exists=], then return [=a promise rejected with=] a {{TypeError}}.
```

Comment:
This prevents `appHistory.navigate()` as some previous discussion on the issue tracker indicated that it probably indicates programmer error.